### DISCUSSION TITLE
The file system does not support extended attributes or has the feature disabled.

### DISCUSSION THREAD

**Discussion by JCKodel** [supabase/supabase#38608] [Questions] [answered]
# Bug report

- [X] I confirm this is a bug with Supabase, not with my own application. 
- [X] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

File uploading through Studio is issuing error 500:

<img width="661" alt="image" src="https://github.com/user-attachments/assets/8d6001f5-a73d-452d-bf54-24d449d97a6e">

Looking the issues here, you guys recommend to enable the VirtioFS option in Docker, but this option was enabled already:

<img width="804" alt="image" src="https://github.com/user-attachments/assets/bdfb61b8-535d-45bf-87dd-8e60652c7e56">

Using MacOS Sequoia 15.0.1

## To Reproduce

I just followed the example listed in https://supabase.com/docs/guides/self-hosting/docker (didn't even touched the .env file, I just run and start to test storage by creating a bucket and dragging an image from my desktop to it).

## Expected behavior

To actual work.

## System information

- OS: MacOS Sequoia 15.0.1
- Browser (if applies) Chrome
- Version of supabase-js: Not sure
- Version of Node.js: Not sure

**Comment by jmuzina** [comment, marked as answer]
I was also having the same issue, uploading files in the dashboard with a stock `.env` throws this error. I'm using macOS 15.0.1 on Firefox, working off of Supabase tag v1.24.09.

@JCKodel try out [this comment](https://github.com/supabase/supabase/issues/20096#issuecomment-1997263218), it has fixed the issue for me.

`docker compose -f docker-compose.yml -f docker-compose.s3.yml up -d`

**Comment by DrKyro** [comment]
> I was also having the same issue, uploading files in the dashboard with a stock `.env` throws this error. I'm using macOS 15.0.1 on Firefox, working off of Supabase tag v1.24.09.
> 
> @JCKodel try out [this comment](https://github.com/supabase/supabase/issues/20096#issuecomment-1997263218), it has fixed the issue for me.
> 
> `docker compose -f docker-compose.yml -f docker-compose.s3.yml up -d`

I try this command，but it still doesn't work properly.
```
  docker git:(master) ✗ docker compose -f docker-compose.yml -f docker-compose.s3.yml up -d
[+] Running 15/16
 ✔ Network supabase_default                  Created                                                                                                                                                                0.0s 
 ⠸ Container supabase-minio-1                Starting                                                                                                                                                               0.4s 
 ✔ Container supabase-minio-createbucket-1   Created                                                                                                                                                                0.0s 
 ✔ Container supabase-vector                 Started                                                                                                                                                                0.4s 
 ✔ Container supabase-imgproxy               Started                                                                                                                                                                0.4s 
 ✔ Container supabase-db                     Created                                                                                                                                                                0.0s 
 ✔ Container supabase-analytics              Created                                                                                                                                                                0.0s 
 ✔ Container supabase-rest                   Created                                                                                                                                                                0.1s 
 ✔ Container supabase-edge-functions         Created                                                                                                                                                                0.1s 
 ✔ Container supabase-studio                 Created                                                                                                                                                                0.1s 
 ✔ Container supabase-meta                   Created                                                                                                                                                                0.1s 
 ✔ Container supabase-auth                   Created                                                                                                                                                                0.1s 
 ✔ Container supabase-kong                   Created                                                                                                                                                                0.1s 
 ✔ Container realtime-dev.supabase-realtime  Created                                                                                                                                                                0.1s 
 ✔ Container supabase-pooler                 Created                                                                                                                                                                0.1s 
 ✔ Container supabase-storage                Created                                                                                                                                                                0.0s 
Error response from daemon: network 5dd09c632b9dd5af12b30ed99dc08cf88136d17df6183c34abc19af19798ee2b not found
➜  docker git:(master) ✗ 
```

**Comment by Hallidayo** [comment]
Hi all, Are you still experiencing this issue?

**Comment by baptiste-mnh** [comment]
Hello,

Same error on my side with the exact same configuration.
Just used the `docker/reset.sh` and the .env.exemple file.

**Comment by aantti** [comment]
If it helps - I've recently bumped into this too.
I guess, this is because the "file system" backend for Storage uses xattr. See here - 
https://github.com/supabase/storage/blob/23559aaabeba90dd0adaa30a548ec0c3322f793b/src/storage/backend/file.ts#L570,
which doesn't work in Docker on macOS.
The workaround with docker-compose.s3.yml helps because it uses minio.

**Comment by Hallidayo** [comment]
Hi all, I'm going to move this over as a discussion for others to use for help.
